lib/swaps/Swaps.ts

```diff
@@ -150,6 +165,40 @@ class Swaps extends EventEmitter {
     return;
   }
 
+  /**
+   * Saves deal to database and deletes from memory.
+   * @param deal The deal to persist.
+   */
+  private persistDeal = (deal: SwapDeal) => {
```

Comment:
I kind of prefer `archiveDeal` or something to that effect. That makes it more clear that we are removing the deal from memory, persist sounds like we might only be saving it.